### DISCUSSION TITLE
Animating color shifts when column values change

### DISCUSSION THREAD

**Discussion by kevincianfarini** [patrykandpatrick/vico#643] [Ideas]
Hi! I am so pleased with the recent addition to Vico allowing us to theme individual columns of a chart using a ColumnProvider. One thing we do is color our columns based on the magnitude of the Y value. If animating between different values of Y, we might be switching from one color to another. See the below gif: 

![untitled](https://github.com/patrykandpatrick/vico/assets/13242326/f423c216-805a-426d-951d-dd06a5772864)

Animating between the green and the red color shown in the gif would be a wonderful addition. Compose makes this easy with utilities like [`animateColorAsState`](https://developer.android.com/reference/kotlin/androidx/compose/animation/package-summary#animateColorAsState(androidx.compose.ui.graphics.Color,androidx.compose.animation.core.AnimationSpec,kotlin.String,kotlin.Function1)). Perhaps it's worth looking there for inspiration if you decide this is a worthwhile feature 😄

**Comment by patrickmichalik** [comment]
Hello! We’re glad to hear that you’ve found `ColumnCartesianLayer.ColumnProvider` helpful. We’ll be looking into adding the proposed functionality. Thanks for the suggestion!
